mesonbuild/modules/gnome.py

```diff
@@ -1055,47 +1056,93 @@ def yelp(self, state: 'ModuleState', args: T.Tuple[str, T.List[str]], kwargs: 'Y
                 raise MesonException('Yelp requires a list of sources')
         elif args[1]:
             mlog.warning('"gnome.yelp" ignores positional sources arguments when the "sources" keyword argument is set')
-        source_str = '@@'.join(sources)
+        sources_files = [mesonlib.File.from_source_file(state.environment.source_dir,
+                                                        os.path.join(state.subdir, 'C'),
+                                                        s) for s in sources]
 
         langs = kwargs['languages']
+        if not langs:
+            langs = read_linguas(os.path.join(state.environment.source_dir, state.subdir))
+
+        media = kwargs['media']
+        symlinks = kwargs['symlink_media']
+        targets: T.List['Target'] = []
```

Comment:
This isn't right, you put a Data in here just a little down the way, at the very least this should be `T.List[T.Union['Target', build.Data]]`

Reply:
Thanks, I also added `build.SymlinkData`